Activities: More player/screen mixup fixes, Decision Day typo.

MageKing17 opened the pull request:
`CameraMan:GetScrollTarget()` and `CameraMan:SetScrollTarget()` both take a screen index but were often being passed player indices. I've hopefully caught and fixed all such instances. In addition, when `Activity:ScreenOfPlayer()` was being called multiple times in a row, I stored the result of a single call and replaced the multiple calls with that local variable.

Lastly, when I changed Decision Day to use each player's brain position instead of always using Player 1's, I accidentally wrote `pos` instead of `Pos`, so the first argument would've always been `nil`.

MageKing17 committed:
Activities: More player/screen mixup fixes, Decision Day typo.

`CameraMan:GetScrollTarget()` and `CameraMan:SetScrollTarget()` both take a screen index but were often being passed player indices. I've hopefully caught and fixed all such instances. In addition, when `Activity:ScreenOfPlayer()` was being called multiple times in a row, I stored the result of a single call and replaced the multiple calls with that local variable.

Lastly, when I changed Decision Day to use each player's brain position instead of always using Player 1's, I accidentally wrote `pos` instead of `Pos`, so the first argument would've always been `nil`.

Causeless commented on CHANGELOG.md:
```diff
@@ -35,14 +35,14 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 - Fixed an issue where palette index 255 was incorrectly showing as black.
 
+- Fixed instances of `CameraMan:GetScrollTarget()` and `CameraMan:SetScrollTarget()` supplying a player index instead of a screen index.
```
Please try to write the symptom that this is fixing, not the cause without context - this is a bit meaningless to anyone playing the game

MageKing17 committed:
Correct changelog, avoid out-of-bounds array access.

In updating to changelog to actually include an example of the effects of mixing up screen and player indices, I realized that there was no protection against out-of-bounds array accesses in these functions, so I fixed that.

Causeless commented on Source/Managers/CameraMan.cpp:
```diff
@@ -64,10 +64,16 @@ void CameraMan::SetScroll(const Vector& center, int screenId) {
 }
 
 Vector CameraMan::GetScrollTarget(int screenId) const {
+	if (screenId < 0 || screenId >= c_MaxScreenCount)
```
no bracketless ifs

Causeless commented on Source/Managers/CameraMan.cpp:
```diff
@@ -64,10 +64,16 @@ void CameraMan::SetScroll(const Vector& center, int screenId) {
 
 void CameraMan::SetScrollTarget(const Vector& targetCenter, float speed, int screenId) {
+	if (screenId < 0 || screenId >= c_MaxScreenCount)
+		return;
```
no bracketless ifs

MageKing17 replied:
I copied it from `GameActivity::DrawGUI()`, but my preference is for brackets anyway.

MageKing17 committed:
Add brackets around new conditionals.
